internal/controller/pkg/resolver/reconciler.go

```diff
@@ -323,28 +349,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			return reconcile.Result{}, errors.Wrap(r.client.Status().Update(ctx, lock), errCannotUpdateStatus)
 		}
 
-		var pack v1.Package
-		switch dep.Type {
-		case v1beta1.ConfigurationPackageType:
-			pack = &v1.Configuration{}
-		case v1beta1.ProviderPackageType:
-			pack = &v1.Provider{}
-		case v1beta1.FunctionPackageType:
-			pack = &v1.Function{}
-		default:
-			log.Debug(errInvalidPackageType)
-			return reconcile.Result{}, nil
-		}
-
-		pack.SetName(xpkg.ToDNSLabel(ref.Context().RepositoryStr()))
-
-		format := packageTagFmt
-		if strings.HasPrefix(addVer, "sha256:") {
-			format = packageDigestFmt
+		pack, err := NewPackage(dep, addVer, ref)
+		if err != nil {
+			log.Debug(errCreateDependency, "error", err)
```

Comment:
It would be good to use a different error string. Otherwise, it is not clear whether it failed here or below when client.Create is called.